tests/functional/adapter/test_optional_iceberg.py

```diff
@@ -0,0 +1,232 @@
+"""
+Run optional dbt functional tests for Iceberg incremental merge, including delete and incremental predicates.
+
+"""
+import re
+
+import pytest
+
+from dbt.contracts.results import RunStatus
+from dbt.tests.adapter.incremental.test_incremental_predicates import (
+    BaseIncrementalPredicates,
+)
+from dbt.tests.adapter.incremental.test_incremental_unique_id import (
+    BaseIncrementalUniqueKey,
+    models__duplicated_unary_unique_key_list_sql,
+    models__empty_str_unique_key_sql,
+    models__empty_unique_key_list_sql,
+    models__expected__one_str__overwrite_sql,
+    models__expected__unique_key_list__inplace_overwrite_sql,
+    models__no_unique_key_sql,
+    models__nontyped_trinary_unique_key_list_sql,
+    models__not_found_unique_key_list_sql,
+    models__not_found_unique_key_sql,
+    models__str_unique_key_sql,
+    models__trinary_unique_key_list_sql,
+    models__unary_unique_key_list_sql,
+    seeds__add_new_rows_sql,
+    seeds__duplicate_insert_sql,
+    seeds__seed_csv,
+)
+from dbt.tests.util import run_dbt
+
+seeds__expected_incremental_predicates_csv = """id,msg,color
+3,anyway,purple
+1,hey,blue
+2,goodbye,red
+2,yo,green
+"""
+
+seeds__expected_delete_condition_csv = """id,msg,color
+1,hey,blue
+3,anyway,purple
+"""
+
+seeds__expected_predicates_and_delete_condition_csv = """id,msg,color
+1,hey,blue
+1,hello,blue
+3,anyway,purple
+"""
+
+
+class TestIcebergMergeNonUniqueLocation:
```

Comment:
did you consider to put this into test_basic_iceberg.py? I think that checking for not unique location is a pretty core check to do, and I won't put in this test suite - but rather I will put in the main suite for iceberg. WDYT?